### PR TITLE
Check exemplar timestamp before matching

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

This pr adds a timestamp check for exemplars Select API.

If the newest exemplar's timestamp in one series is smaller than the select start time, we can skip this series before label matching. Same for the oldest exemplar.

### CODE REVIEW

File: tsdb/exemplar.go [hunk 1]
```diff
@@ -121,6 +121,9 @@ func (ce *CircularExemplarStorage) Select(start, end int64, matchers ...[]*label
 	for _, idx := range ce.index {
 		var se exemplar.QueryResult
 		e := ce.exemplars[idx.oldest]
+		if e.exemplar.Ts > end || ce.exemplars[idx.newest].exemplar.Ts < start {
```
Comment (roidelapluie) [inline tsdb/exemplar.go:124]:
```suggestion
		if e.exemplar.Ts >= end || ce.exemplars[idx.newest].exemplar.Ts <= start {
```

**Review by roidelapluie — COMMENTED** [review]
cc @cstyan for a deeper review

File: tsdb/exemplar.go (hunk 1 above)
Comment (yeya24) [reply tsdb/exemplar.go:124]: If `>=`, we still have one exemplar match? Can you please elaborate the reason to change this?

**Review by bwplotka — APPROVED** [review]
Makes sense, thanks!

**Comment by cstyan** [comment]
@yeya24 nice, thank you!

File: tsdb/exemplar.go (hunk 1 above)
Comment (roidelapluie) [reply tsdb/exemplar.go:124]: Don't worry I misread ;)

**Pull request merged**